lib/config/compat_provisioner.rb

```diff
@@ -0,0 +1,73 @@
+require 'teracy-dev'
+
+module TeracyDevV05Compat
+  module Config
+    # see: https://www.vagrantup.com/docs/provisioning/
+    class CompatProvisioner < TeracyDev::Config::Configurator
+
+      def configure_common(settings, config)
+        provisioners_settings = settings['default']['compat_provisioners'] ||= []
+        @logger.debug("configure_node: #{provisioners_settings}")
+        provisioners_settings.each do |provisioner_settings|
+          @logger.info("provisioner ignored: #{provisioner_settings}") if provisioner_settings['enabled'] != true
+          next if provisioner_settings['enabled'] != true
+
+          type = provisioner_settings['type']
+          run = 'once'
+          preserve_order = false
+          if !provisioner_settings['run'].nil?
+            run = provisioner_settings['run'] # one of: once, always, or never
+          end
+          if provisioner_settings['preserve_order'] == true
+            preserve_order = true
+          end
+
+          options = provisioner_settings.dup
+
+          ["_id", "type", "enabled", "run", "preserve_order"].each do |key|
+            options.delete(key)
+          end
+
+          if type == 'chef_solo'
+            if !options['cookbooks_path'].nil?
+              options['cookbooks_path'].each_index { |i|
+                extension_relative_paths = options['cookbooks_path'][i].split(File::SEPARATOR)
```

Comment:
just use `teracy-dev-v05-compat` as this is the constant name, you can create a constant var and use it for the processor (commented above) if you want this to work with teracy-dev-core v0.3.

I recommend supporting teracy-dev-core v0.3 and later.